smart_home/include/rooms/room.h

```diff
@@ -0,0 +1,20 @@
+#ifndef SMART_HOME_ROOMS_ROOM_H_
+#define SMART_HOME_ROOMS_ROOM_H_
+
+namespace rooms {
+
+// Rule of Zero
+
+class Room {
+  public:
+    virtual float size() = 0;
```

Comment:
Although there might be valid reasons to declare the room as an abstract class, in your current implementation I see no need to do it. The size is assigned during the creation of the object and it can be safely returned here for all types of rooms.